src/main/java/net/dv8tion/jda/internal/requests/restaction/InviteActionImpl.java

```diff
@@ -67,14 +69,23 @@ public InviteActionImpl deadline(long timestamp)
         return (InviteActionImpl) super.deadline(timestamp);
     }
 
+    @Override
+    public boolean isRestricted()
+    {
+        return restricted;
+    }
+
     @Nonnull
     @Override
     @CheckReturnValue
     public InviteActionImpl setMaxAge(final Integer maxAge)
     {
         if (maxAge != null)
             Checks.notNegative(maxAge, "maxAge");
-
+        if (restricted) {
+            Checks.check(maxAge != null && maxAge >= 1 && maxAge <= MAX_RESTRICTED_AGE,
```

Comment:
Should this be a `Math.max` instead? Discord is temporarily returning 30 day invites until an unknown date before throwing an error

Reply:
No, I prefer telling the user their intentions are not possible. Otherwise, they might believe it to be a bug or introduce unwanted results.